app/controllers/evidence_controller.rb

```diff
@@ -122,11 +122,14 @@ def destroy
           # 2. from the node evidence tab
           # 3. from the evidence show page itself (under node)
           # When using redirect_back in case 3, we find an evidence not found error,
-          # since the evidence does not exist anymore. That's why we check the 'Referer' here:
-          if request.headers['Referer'] == project_node_evidence_url(current_project, @node, @evidence)
+          # since the evidence does not exist anymore. That's why we check the referrer here.
+          referrer_path = URI(request.referrer).tap { |u| u.query = nil }.to_s
+          if !referrer_path || referrer_path == project_node_evidence_url(current_project, @node, @evidence)
             redirect_to project_node_path(current_project, @node), notice: notice
+          elsif referrer_path == project_node_path(current_project, @node)
+            redirect_to referrer_path, notice: notice
           else
-            redirect_back fallback_location: project_node_path(current_project, @node), notice: notice
+            redirect_to "#{referrer_path}?tab=evidence-tab", notice: notice
```

Comment:
if we need to redirect_to, let's try not no build the url ourselves. Can we use `url_for` ?

Reply:
I agree, but the `referrer_path` that we get from the request is a string. I'm open to ideas!

Reply:
Also noticed that if referer had other query params, we are losing them here.

Reply:
We are redirecting anyway so I don't think we should preserver that?